src/tools/f.ts

```diff
@@ -27,43 +29,75 @@ export const padString = (value, expectedMaxLength = '100%'.length) => {
   return value + ' '.repeat(expectedMaxLength - String(value).length);
 }
 
-// Used to print message in channel, use for every use of channel.send for consistency
-// Returns promise for use in async functions
-export const print = function (messageVar, message, callback = callbackPrintDefault) {
-  console.log(`> ${message.embed ? `(Embed)${message.embed.title}\t${message.embed.description}` : message}`);
-  if (message.length >= 2000) {
-    let sent = false;
-    for (let i = 2000; i >= 0; i--) {
-      if (message.charAt(i) === '\n') {
-        const firstMessage = message.substring(0, i);
-        const restMessage = message.substring(i);
-        messageVar.channel.send(firstMessage)
-          .then((result) => {
-            callback(result);
-          }).catch(err => console.log(`@print (splitted) for ${message} :\n${err}`));
-        print(messageVar, restMessage, callback);
-        sent = true;
-        break;
-      }
-    }
-    if (!sent) { // No newline in message, split content on max size
-      const maxIndex = 1900;
-      const firstMessage = message.substring(0, maxIndex); // Margin
-      const restMessage = message.substring(maxIndex);
+/**
+ * Used to print message in channel, use for every use of channel.send for consistency
+ * Returns promise for use in async functions
+ */
+export const print = (messageVar: Message, message: string | EmbeddedMessage, callback = callbackPrintDefault) => {
+  let printableCommand = message;
+  if ((message as EmbeddedMessage).embed) { // Embedded Message
+    const embeddedMessage = message as EmbeddedMessage;
+    printableCommand = `(Embed)${embeddedMessage.embed.title}\t${embeddedMessage.embed.description}`;
+    messageVar.channel.send(message)
```

Comment:
Is it enteded to be `message` here? Just double checking since I haven't grasped the EmbeddedMessage handling fully.

Reply:
Embedded Message is basically just a slimmer version of input for a Message Object, but they had some differences and haven't found the real difference / type that discord uses for it, so its just a interface with 3 fields currently (enough to send the message)